### PR TITLE
Fix 420 comments parsing

### LINKED ISSUE #420: ldapbindpasswd in options which includes  '#' creates faulty pg_hba lines
##### SUMMARY
ldapbindpasswd in options which includes  '#' creates faulty pg_hba lines
While a hashmark in a password might be ugly I cannot change this

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
postgresql_pg_hba

##### ANSIBLE VERSION

```paste below

ansible [core 2.11.12]
  config file = None
  configured module search path = ['/home/USER/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/USER/.local/lib/python3.6/site-packages/ansible
  ansible collection location = /home/USER/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/USER/.local/bin/ansible
  python version = 3.6.8 (default, Aug 13 2020, 07:46:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
  jinja version = 3.0.1
  libyaml = True

```

##### COLLECTION VERSION

```paste below
Collection           Version
-------------------- -------
community.postgresql 2.3.2
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

```yaml
---
- hosts: localhost
  tasks:
    - name: line 1
      community.postgresql.postgresql_pg_hba:
        dest: pg_hba.conf
        create: yes
        users: bug
        databases: bug
        source: 66.66.66.66/32
        contype: hostssl
        method: ldap
        options: ldapbindpasswd ="#BROKEN"
        keep_comments_at_rules: true
        comment: messed up
    - name: line 2 breaks line 1
      community.postgresql.postgresql_pg_hba:
        dest: pg_hba.conf
        create: yes
        users: bug2
        databases: bug
        source: 66.66.66.66/32
        contype: hostssl
        method: ldap
        options: ldapbindpasswd ="#BROKEN"
        keep_comments_at_rules: true
```

##### EXPECTED RESULTS

file pg_hba should be:
```
hostssl bug     bug     66.66.66.66/32  ldap    ldapbindpasswd ="#BROKEN"        #messed up
hostssl bug     bug2    66.66.66.66/32  ldap    ldapbindpasswd ="#BROKEN"
```

##### ACTUAL RESULTS

pg_hba.conf generated is:
```
hostssl bug     bug     66.66.66.66/32  ldap    ldapbindpasswd ="    #BROKEN"        #messed up
hostssl bug     bug2    66.66.66.66/32  ldap    ldapbindpasswd ="#BROKEN"
```

```paste below
PLAY [localhost] *****************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************************************************************************************************************************
ok: [localhost]

TASK [line 1] ********************************************************************************************************************************************************************************************************************************************************************************
--- before
+++ after
@@ -1,4 +1,6 @@
 {
     "file": "pg_hba.conf",
-    "pg_hba": []
+    "pg_hba": [
+        "hostssl\tbug\tbug\t66.66.66.66/32\tldap\tldapbindpasswd =\"#BROKEN\""
+    ]
 }

changed: [localhost]

TASK [line 2 breaks line 1] ******************************************************************************************************************************************************************************************************************************************************************
--- before
+++ after
@@ -1,4 +1,6 @@
 {
     "file": "pg_hba.conf",
-    "pg_hba": []
+    "pg_hba": [
+        "hostssl\tbug\tbug2\t66.66.66.66/32\tldap\tldapbindpasswd =\"#BROKEN\""
+    ]
 }

changed: [localhost]

PLAY RECAP ***********************************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=3    changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

### PR DESCRIPTION
**Pull request by toydarian**
##### SUMMARY

- Fixes issue parsing `pg_hba.conf` files that contain a `#` in a quoted string
- When there is an issue parsing a `pg_hba.conf` file, we now print the number of the line where the issue is
- Adds tests for the modified and added functionality

Fixes #420 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
postgresql_pg_hba

### CODE REVIEW

**Comment by Andersson007** [comment]
@betanummeric could you please take a look when you have a chance?

**Review by hunleyd — APPROVED** [review]

File: tests/integration/targets/postgresql_pg_hba/tasks/postgresql_pg_hba_initial.yml [hunk 1]
```diff
@@ -269,3 +271,88 @@
...
+
+- name: check that parsing failed (too few symbols)
+  postgresql_pg_hba:
+    dest: /tmp/faulty_test_pg_hba.conf
+  register: faulty_pg_hba
+  ignore_errors: true
+
+- assert:
+    that:
+      - faulty_pg_hba is failed
```
Comment (Andersson007) [inline tests/integration/targets/postgresql_pg_hba/tasks/postgresql_pg_hba_initial.yml:358]:
```suggestion
- assert:
    that:
      - faulty_pg_hba is failed
```
Can we use `- faulty_pg_hba is search( "..." )` and a line number of the affected line ? Greg for `search(` in the collection tests for examples

Comment (Andersson007) [reply tests/integration/targets/postgresql_pg_hba/tasks/postgresql_pg_hba_initial.yml:358]: UPDATE: `- faulty_pg_hba is search("...")` -> `- faulty_pg_hba.msg is search("...")`. I forget `msg`

Comment (toydarian) [reply tests/integration/targets/postgresql_pg_hba/tasks/postgresql_pg_hba_initial.yml:362]: I wasn't able to commit your suggestion. Added additional checks in d2873d6

**Comment by Andersson007** [comment]
@toydarian LGTM
i'll wait for @betanummeric 's review until tomorrow and if no objections/response i'll merge it, thanks!

**Comment by Andersson007** [comment]
@toydarian thanks for the great improvement!
@hunleyd thanks for reviewing!

**Comment by stelb** [comment]
I was just thinking yesterday if I could look into this issue again.. thank you! :)

**Pull request merged**